### ISSUE TITLE
XGBClassifier always stopping after 99 rounds

### ISSUE THREAD

**Issue by Padu0827** [dmlc/xgboost#4565]
### Issue
While training a model using XGBClassifier with early stopping in Python, the model always stops after 99 training iterations unless early stopping is set to something less than that. I'd like to  train this model for many more iterations than just 99, but no matter what I set early_stopping_rounds to it stops after 99.

### Source
downloaded through Anaconda

### Code
```
eval_set = [(X_test, y_test)]
print('Training')
xgb = XGBClassifier(max_depth=2, learning_rate=0.01, objective='binary:logistic', random_state=seed, verbose=2)

xgb.fit(X_train, y_train, early_stopping_rounds=200, eval_metric='auc', eval_set=eval_set, verbose=True)
```

**Comment by hcho3** [comment]
How does the AUC metric change over time? Maybe early stopping is working as intended?
